### ISSUE TITLE
server: Log spam about invalid security handshakes from localhost during startup

### ISSUE THREAD

**Issue by a-robinson** [cockroachdb/cockroach#20929]
On cyan-0001, during its most recent startup there are 6900 copies of the message below (from different source ports) in the first 7 seconds of the node being up before they stop:

```
W171220 15:00:42.061583 74 vendor/google.golang.org/grpc/server.go:507  grpc: Server.Serve failed to complete security handshake from "127.0.0.1:57088": tls: first record does not look like a TLS handshak
e
```

Their stop happens at the same time as the first `started gossip client` message, so that may be related.

@tschottdorf @mberhault

**Comment by mberhault** [comment]
Yeah, I saw those while poking at #20928 but filed it away for later. I'll take a look. Since the nodes were already initialized, we should be jumping straight into full server mode, right?

**Comment by a-robinson** [comment]
It shouldn't be waiting around in the `Init` state unless it lost its data, correct.

**Comment by benesch** [comment]
I just saw this on omega-3 while deploying the latest alpha (https://github.com/cockroachdb/cockroach/issues/21480), SHA https://github.com/cockroachdb/cockroach/commit/9da836000c7cc1ca058a4082a80c741c942b58c3. Tons of TLS handshake logspam, followed by an exit. Starting the node a second time produced logspam, but less, and then everything went back to normal.

¯\_(ツ)_/¯

**Comment by bdarnell** [comment]
This was fixed by #21682
